tests.sh

```diff
@@ -0,0 +1,182 @@
+#!/bin/bash
+
+set -e
+
+if [ -z $1 ]; then
+    echo "Stack name not passed, exiting"
+    exit 1
+fi
+
+STACK=$1
+echo "$STACK"
+
+
+function clean {
+
+    ## Cleanup DB
+
+
+    for f in "${TEST_FILES[@]}"; do
+
+        FILE_NAME=$(cut -d "/" -f2- <<< "$f")
+
+        echo "Removing item from DB : $FILE_NAME"
+
+        aws dynamodb delete-item --table-name $DYNAMO_TABLE \
+        --key "{\"id\": {\"S\": \"s3://$BUCKET_IN/$FILE_NAME\"}}"
+
+
+        if [ $? -ne 0 ]; then
+            echo -e "Remove item failed" ; exit 1
+        fi
+
+    done
+
+    ## Cleanup Output
+
+
+    for f in "${TEST_FILES[@]}"; do
+
+      ## We know our output files will have a html extention so we need to substitute md for html when working with the output.   
+
+      FILE_NAME=$(cut -d "/" -f2- <<< "$f")
+
+      echo "Removing from output : ${FILE_NAME/md/html}"
+
+      aws s3 rm s3://"$BUCKET_OUT"/"${FILE_NAME/md/html}"
+
+        if [[ $? -ne 0 ]]; then
+          echo -e "File not processed" ; exit 1
+        fi
+
+    done
+
+    ## Cleanup Input
+
+
+    for f in "${TEST_FILES[@]}"; do
+
+      FILE_NAME=$(cut -d "/" -f2- <<< "$f")
+
+      echo "Removing from input: $FILE_NAME"
+
+      aws s3 rm s3://"$BUCKET_IN"/"$FILE_NAME"
+
+        if [[ $? -ne 0 ]]; then
+          echo -e "File not processed" ; exit 1
+        fi
+
+
+    done
+
+
+}
+
+
+## Get Stack Resources
+
+BUCKET_IN=$(aws cloudformation describe-stack-resource \
+--stack-name "$STACK" --logical-resource-id 'InputBucket' \
+--query "StackResourceDetail.PhysicalResourceId" \
+--output text) || echo -e "Couldn't get stack details" ; exit 1
+
+BUCKET_OUT=$(aws cloudformation describe-stack-resource --stack-name "$STACK" \
+--logical-resource-id 'ConversionTargetBucket' \
+--query "StackResourceDetail.PhysicalResourceId" \
+--output text) || echo -e "Couldn't get stack details" ; exit 1
```

Comment:
I had to use the following syntax to:
`--output text) || ( echo -e "Couldn't get stack details" && exit 1 )` 

Otherwise the exit would always run.